graphql_compiler/compiler/emit_sql.py

```diff
@@ -205,81 +259,120 @@ def _set_group_by(self, group_by):
         """Set output columns for the fold object."""
         self._group_by = group_by
 
-    def _construct_fold_joins(self, edge, from_alias, to_alias):
-        """Use the edge descriptors to create the join clause between the tables in the fold."""
-        join_clause = sqlalchemy.join(
-            from_alias,
-            to_alias,
-            onclause=(from_alias.c[edge.from_column] == to_alias.c[edge.to_column])
-        )
+    def _construct_fold_joins(self):
+        """Use the traversal descriptors to create the join clause for the tables in the fold."""
+        # Start the join clause with the from_table of the first traversal descriptor,
+        # which is the vertex immediately preceding the fold
+        join_clause = self._traversal_descriptors[0].from_table
+
+        # MSSQL and PostgreSQL have different terminating indices
+        terminating_index = len(self._traversal_descriptors)
+        traversal_descriptors = self._traversal_descriptors[:terminating_index]
+
+        # Starting at the first from_table, join traversed vertices in order until the output
+        # vertex (PostgreSQL) is reached
+        for travel_descriptor in traversal_descriptors:
+            # joins from earlier in the chain of traversals are at the beginning of the list
+            # b/c joins are appended in the order they are traversed
+            from_table = travel_descriptor.from_table
+            to_table = travel_descriptor.to_table
+            join_descriptor = travel_descriptor.join_descriptor
+            join_clause = sqlalchemy.join(
+                join_clause,
+                to_table,
+                onclause=(from_table.c[join_descriptor.from_column] == to_table.c[
+                    join_descriptor.to_column
+                ])
+            )
```

Comment:
This is just a generalization of the original logic. It isn't quite a fully functional implementation of multiple traversals, because that requires changing some other stuff in the compilation state.

Can I leave this loop as is but put a NotImplementedError at the beginning of the method if traversal_descriptors has more than 1 element?